plugins/woocommerce/client/legacy/js/admin/system-status.js

```diff
@@ -112,6 +113,17 @@ jQuery( function ( $ ) {
 		copyFail: function() {
 			$( '.copy-error' ).removeClass( 'hidden' );
 			$( '#debug-report' ).find( 'textarea' ).trigger( 'focus' ).trigger( 'select' );
+		},
+
+		downloadReport: function() {
+			var ssr_text = new Blob( [ $( '#debug-report' ).find( 'textarea' ).val() ], { type: 'text/plain' } );
+			var a = document.createElement( 'a' );
+			a.download = 'SystemStatusReport.txt';
```

Comment:
Suggestion: maybe the file name should be unique each time? Two reasons:
 - We help the user figure out which one is the newest in a potential sea of "SystemStatusReport.txt", "SystemStatusReport (1).txt", etc. Depending on the filesystem and how duplicate files are handled, for example `SystemStatusReport (1).txt` might be sorted next to `SystemStatusReport (10).txt` and the user could get further confused.
 - We help our HEs figure out when the SSR was downloaded and how "fresh" it is (I did not see a "current date/time" field in the SSR - please let me know if I missed it).

So something like `SystemStatusReport - 2023-05-04 1024.txt`. It should also naturally sort itself by name, so hopefully the user will pick the newest one.

I'm not sure how often that happens, but maybe also adding the site domain to the filename could help, if the user has more than one site.

Reply:
Thanks for the suggestion, I think it makes sense. Added in [6cfdf13](https://github.com/woocommerce/woocommerce/pull/38110/commits/6cfdf1313c6ce702d18bfd543c53373f2190a585)